src/NuGet.Indexing/DocumentCreator.cs

```diff
@@ -264,14 +264,32 @@ static void AddSupportedFrameworks(Document document, IDictionary<string, string
                 }
             }
         }
+        static void AddRequiredField(Document document, string destName, IDictionary<string, string> package, List<string> errors, string sourceName, Field.Index fieldIndex, float boost = 1.0f)
+        {
+            if (!AddField(document, destName, package, sourceName, fieldIndex, boost))
+            {
+                errors.Add($"Required property '{sourceName}' not found.");
+            }
+        }
+
+        static void AddFieldWithDefault(Document document, string destName, IDictionary<string, string> package, string sourceName, string defaultValue, Field.Index fieldIndex, float boost = 1.0f)
+        {
+            if (!AddField(document, destName, package, sourceName, fieldIndex, boost))
+            {
+                document.Add(new Field(destName, defaultValue, Field.Store.YES, fieldIndex) { Boost = boost });
+            }
+        }
 
-        static void AddField(Document document, string destName, IDictionary<string, string> package, string sourceName, Field.Index fieldIndex, float boost = 1.0f)
+        static bool AddField(Document document, string destName, IDictionary<string, string> package, string sourceName, Field.Index fieldIndex, float boost = 1.0f)
         {
             string value;
             if (package.TryGetValue(sourceName, out value))
```

Comment:
`null` will be rejected by rejected by Lucene so adding additional validation is not necessary. `String.Empty` is an acceptable input.